### PR TITLE
Fix ItemRetrieval assertion and output errors

### PR DESCRIPTION
**Pull request by rnyak**
- This fix the error reported here https://github.com/NVIDIA-Merlin/models/issues/173

- we also find another issues in the shape of `positive_item_ids`  and fixed it.

### CODE REVIEW

File: merlin/models/tf/core.py [hunk 1]
```diff
@@ -730,7 +730,7 @@ def compute_loss(self, inputs, targets, **kwargs):
     def call_targets(self, predictions, targets, training=False, **kwargs):
         outputs = targets
         for layer in self.layers:
-            if isinstance(outputs, tuple):
+            if isinstance(outputs, tuple) and len(outputs) > 0:
```
Comment (marcromeyn) [inline merlin/models/tf/core.py:733]:
```suggestion
            if isinstance(outputs, (list, tuple)) and len(outputs) > 0:
```

File: merlin/models/tf/prediction/item_prediction.py [hunk 2]
```diff
@@ -143,6 +143,8 @@ class ItemRetrievalScorer(Block):
     item_id_feature_name: str
         Name of the column containing the item ids
         Defaults to `item_id`
+    query_name: str = 'query'
```
Comment (marcromeyn) [inline merlin/models/tf/prediction/item_prediction.py:146]: Could use a single line of explanation, same for `item_name`

File: merlin/models/tf/prediction/item_prediction.py [hunk 3]
```diff
@@ -179,7 +186,7 @@ def add_features_to_context(self, feature_shapes) -> List[str]:
         return self._required_features
 
     def _check_input_from_two_tower(self, inputs):
-        assert set(inputs.keys()) == {"query", "item"}
+        assert set(inputs.keys()) == set([self.query_name, self.item_name])
```
Comment (marcromeyn) [inline merlin/models/tf/prediction/item_prediction.py:189]:
```suggestion
        if set(inputs.keys()) != set([self.query_name, self.item_name]):
            raise ValueError(f“Wrong input-names, expected: {[self.query_name, self.item_name]} but got: {inputs.keys()}")
```

**Review by marcromeyn — CHANGES_REQUESTED** [review]
Looks good, I left some small comments

File: merlin/models/tf/prediction/item_prediction.py (hunk 2 above)
Comment (rnyak) [reply merlin/models/tf/prediction/item_prediction.py:146]: yes for sure.

File: merlin/models/tf/core.py (hunk 1 above)
Comment (rnyak) [reply merlin/models/tf/core.py:733]: thanks.

**Review by gabrielspmoreira — APPROVED** [review]
Looks good to me. 
Have included here fixes to deal with the zero loss (bug in the ItemRetrievalScorer) and nan loss when using mixed precision (fp16) due to min_float being smaller than the min float32 value.

**Review by marcromeyn — APPROVED** [review]

**Pull request merged**